sys/include/stdio_semihosting.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2020 Koen Zandberg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_stdio_semihosting STDIO over Semihosting
+ * @ingroup     sys
+ *
+ * @brief       Standard input/output backend using ARM Semihosting
+ *
+ * The ARM Semihosting provides an STDIO backend using the ARM Semihosting
+ * protocol. The main advantage of Semihosting is that is allows STDIO over the
+ * SWD/JTAG debugging interface already available on ARM microcontrollers.
+ *
+ * ARM Semihosting works by using the breakpoint instructing to trigger the
+ * debugger to read the output or to write the input chars. Please be aware that
+ * this might skew the timing of your application.
+ *
+ * The main disadvantage of Semihosting is that it is relative slow (even when
+ * compared to serial uart), and that it requires an active debug session to
+ * handle the breakpoint instructions. Without an active debug session the CPU
+ * will halt on the first STDIO activity until the breakpoint is handled by the
+ * debugger. Don't forget to disable the Semihosting module or replace it with
+ * stdio_null when switching to production builds.
+ *
+ * As this is an ARM specific protocol, this module will only work on ARM-based
+ * microcontrollers.
+ *
+ * ## Usage
+ *
+ * Enable Semihosting-based stdio by adding the following module to your
+ * makefile:
+ *
+ * ```
+ * USEMODULE += stdio_semihosting
+ * ```
+ *
+ * If semihosting is not the default stdio mechanism of your board, the
+ * `RIOT_TERMINAL` variable has to be set to `semihosting`:
+ *
+ * ```
+ * make term RIOT_TERMINAL=semihosting
```

Comment:
You didn't add this to `Makefile.include` because of inclusion order right?

Reply:
You mean to automatically setting the `RIOT_TERMINAL` based on whether `stdio_semihosting` is enabled? I honestly haven't tried it :D

Reply:
It would work if `USEMODULE` is passe by the user or in the `APPLICATION` but not in a `BOARD` configuration, which kind of sucks.